src/QAT/qat/purr/compiler/instructions.py

```diff
@@ -337,28 +337,27 @@ def __init__(
         output_variable=None,
         existing_names: Set[str] = None,
         delay=None,
-        filter: Pulse = None,
+        filter: Union[Pulse, CustomPulse] = None
```

Comment:
same as `Waveform`

Reply:
Assuming it was Pulse so AbstractWaveform couldn't be used? I actually have no ideal what AbstractWaveform is used for, but happy to change this to Waveform

Reply:
`Waveform` is the right type information that satisfies needs expressed here as its descendents are exactly `Pulse` and `CustomPulse`

Reply:
I agree `AbstractWaveform` is a little confusing ... but that's another issue ^^